### ISSUE TITLE
Support optionalKeys

### ISSUE THREAD

**Issue by thomasmikava** [hjkcai/typesafe-joi#4]
Is it theoretically possible to support optionalKeys event partially in typescript?

I was trying something like this, but it does not work.

```ts
optionalKeys<Map extends SchemaMap, K extends {[key in keyof T]: key extends keyof Map ? Map[key]: any}, T extends keyof Map> (children: T[]): 
  SchemaType<Schema, (
	{ [key in keyof SchemaMapLiteral<FilterOut<Map, K>>]: SchemaMapLiteral<FilterOut<Map, K>>[key] }
	& { [key in keyof SchemaMapLiteral<Filter<Map, K>>]?: SchemaMapLiteral<Filter<Map, K>>[key] }
  )>
```

Is there a possibility of supporting optionalKeys?

**Comment by hjkcai** [comment]
Theoretically, yes. But what if one of the arguments is a variable which has the type `string` other than string literal type? I guess all keys will be marked as optional by typescript.

**Comment by thomasmikava** [comment]
Well, it will certainly be limitation

**Comment by thomasmikava** [comment]
Can't we check that by something similar to this
```ts
key extends keyof something ? then_make_optional : otherwise_not 
```
?
Similar implementation will be for forbiddenKeys too

**Comment by hjkcai** [comment]
I am working on 2.0.0 and it will be easier to do this. I will have a try 😀

**Comment by hjkcai** [comment]
@thomasmikava operations of object keys have been implemented in typesafe-joi 2.0.0-alpha6. Please have a try! If you found any bug, feel free to open another issue 😀

```
npm i typesafe-joi@next
```
